packages/google_sign_in/google_sign_in_ios/ios/Classes/FLTGoogleSignInPlugin.m

```diff
@@ -84,32 +84,45 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
                                  message:@"Games sign in is not supported on iOS"
                                  details:nil]);
     } else {
+      NSMutableDictionary<NSString *, NSString *> *plist = nil;
       NSString *path = [[NSBundle mainBundle] pathForResource:@"GoogleService-Info"
                                                        ofType:@"plist"];
       if (path) {
-        NSMutableDictionary<NSString *, NSString *> *plist =
-            [[NSMutableDictionary alloc] initWithContentsOfFile:path];
-        BOOL hasDynamicClientId = [call.arguments[@"clientId"] isKindOfClass:[NSString class]];
-
-        if (hasDynamicClientId) {
-          self.signIn.clientID = call.arguments[@"clientId"];
-        } else {
-          self.signIn.clientID = plist[kClientIdKey];
+        plist = [[NSMutableDictionary alloc] initWithContentsOfFile:path];
+      }
+
+      BOOL hasDynamicClientId = [call.arguments[@"clientId"] isKindOfClass:[NSString class]];
+      if (hasDynamicClientId) {
+        self.signIn.clientID = call.arguments[@"clientId"];
+      } else {
+        if (!plist) {
+          result([FlutterError errorWithCode:@"missing-config"
+                                     message:@"GoogleService-Info.plist file not found"
+                                     details:nil]);
+          return;
         }
+        self.signIn.clientID = plist[kClientIdKey];
+      }
 
+      BOOL hasDynamicServerClientId =
+          [call.arguments[@"serverClientId"] isKindOfClass:[NSString class]];
+      if (hasDynamicServerClientId) {
+        self.signIn.serverClientID = call.arguments[@"serverClientId"];
+      } else if (plist) {
         self.signIn.serverClientID = plist[kServerClientIdKey];
-        self.signIn.scopes = call.arguments[@"scopes"];
-        if (call.arguments[@"hostedDomain"] == [NSNull null]) {
-          self.signIn.hostedDomain = nil;
-        } else {
-          self.signIn.hostedDomain = call.arguments[@"hostedDomain"];
-        }
-        result(nil);
       } else {
-        result([FlutterError errorWithCode:@"missing-config"
-                                   message:@"GoogleService-Info.plist file not found"
-                                   details:nil]);
+        self.signIn.serverClientID = nil;
```

Comment:
This seems defer from the current behavior, do we not need to return an error if serverClientID is missing?

Reply:
It does change the behavior to making `serverClientID` optional. It is not required by the Google Sign-In iOS SDK:
> If you need to get users' ID tokens for [backend authentication](https://developers.google.com/identity/sign-in/ios/backend-auth), also set the serverClientID parameter.

https://developers.google.com/identity/sign-in/ios/sign-in#configure

Reply:
Actually, after looking at it again, I realized that, `serverClientID` was never a required value in the config file. If it did not exist, it just initialized the iOS SDKs `serverClientID` to `nil`. That behavior is kept. So, I havn't changed the CHANGELOG, but have expanded the docs for `clientId` and `serverClientId` in the Dart init method.